packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -76,21 +76,33 @@ class ConfigManager implements IConfigManager {
     updateStorageStateFromLoadOptions(this.logger);
     updateConsentsStateFromLoadOptions(this.logger);
 
+    const { useServerSideCookies, dataServerUrl, logLevel, configUrl } = state.loadOptions.value;
+
     // set application lifecycle state in global state
     batch(() => {
       state.lifecycle.integrationsCDNPath.value = intgCdnUrl;
       state.lifecycle.pluginsCDNPath.value = pluginsCDNPath;
 
-      if (state.loadOptions.value.logLevel) {
-        state.lifecycle.logLevel.value = state.loadOptions.value.logLevel;
+      if (logLevel) {
+        state.lifecycle.logLevel.value = logLevel;
       }
 
       state.lifecycle.sourceConfigUrl.value = getSourceConfigURL(
-        state.loadOptions.value.configUrl,
+        configUrl,
         state.lifecycle.writeKey.value as string,
         lockIntegrationsVersion,
         this.logger,
       );
+
+      if (useServerSideCookies && dataServerUrl) {
+        const isValidUrl = isValidDataServerUrl(dataServerUrl, this.logger);
```

Comment:
`isValidUrl` should also be used to determine if the feature should be disabled.

Reply:
isValidDataServerUrl method uses isValidUrl to determine if this is a valid url